proposed/middleware-meta.md

```diff
@@ -0,0 +1,133 @@
+PSR-N Meta Document
+===================
+
+1. Summary
+----------
+
+The purpose of this PSR is to provide an interface that defines the formal
+method signature of a PSR-7 compliant HTTP middleware component based on
+existing informal standards.
+
+2. Why Bother?
+--------------
+
+The general concept of a reusable request/response middleware was present in
+[StackPHP][stackphp]. Since the release of PSR-7, a number of frameworks have
+applied the StackPHP concept with a different interface signature. The signature
+for these middleware implementations has been almost universally the same.
+
+Agreeing on a formal middleware interface eliminates several problems and
+provides a number of benefits:
+
+* Provides a formal standard for middleware developers to commit to.
+* Eliminates duplication of similar interfaces defined by various frameworks.
+* Avoids minor discrepancies in method signatures.
+* Enables any middleware component to run in any compatible framework.
+
+Although many frameworks already use a similar signature, a formal PSR will
+help middleware and framework vendors to formally certify compliance.
+
+[stackphp]: http://stackphp.com/
+
+3. Scope
+--------
+
+## 3.1 Goals
+
+* Provide a middleware interface that is compatible with PSR-7 as a Composer package.
```

Comment:
Maybe mention "PSR-7 Http Messages" to keep it clear exactly whats happening. Lots of numbers can confuse.